### DISCUSSION TITLE
New token for every request

### DISCUSSION THREAD

**Discussion by jrvc1919** [prkumar/uplink#260] [Q&A] [answered]
Hello 👋 
I have a consumer with several requests and need to add a new jwt token as a Header to each new request.
I have a function that returns a new token. Let's call it `get_jwt()`.
How can I generate a new token, and add it to each new request in runtime?
Without changing the session token (as I am dealing with concurrency)
Thank you!

**Comment by prkumar** [comment, marked as answer]
Hey!

One option is to use the [Header](https://uplink.readthedocs.io/en/latest/dev/types.html?highlight=Header#header) argument annotation. 

If you need to prepend "Bearer" to the token returned by `get_jwt`, you could wrap the call. For example:

```python
class MyConsumer(Consumer):

    def get_user(self):
        return self._get_user(f"Bearer {get_jwt()}")

    @get("user")
    def _get_user(self, token: Header("Authorization")):
        pass
```

**Reply by jrvc1919** [reply]
Thanks for you reply!
When I try to do that I get `unexpected argument` for the "auth" argument.\
I have tried changing to "Authorization" like the documentation.\
What am I missing?

**Reply by prkumar** [reply]
Whoops, yup that should be "Authorization". Hmm.. could you share the traceback?

**Reply by jrvc1919** [reply]
Sorry @prkumar it was just a linting issue 😅  My bad
Thank you for your help!!!!

**Reply by prkumar** [reply]
No worries! Happy to help!
